Mathlib/LinearAlgebra/ExteriorAlgebra/Basic.lean

```diff
@@ -417,6 +417,58 @@ theorem ι_range_map_map (f : M →ₗ[R] N) :
     Submodule.map (ι R) (LinearMap.range f) :=
   CliffordAlgebra.ι_range_map_map _
 
+def mapTriv [Module Rᵐᵒᵖ M] [IsCentralScalar R M] [Module Rᵐᵒᵖ N] [IsCentralScalar R N]
+    (f : M →ₗ[R] N) : TrivSqZeroExt R M →ₐ[R] TrivSqZeroExt R N := by
+  apply TrivSqZeroExt.lift (TrivSqZeroExt.inlAlgHom R R N) ((TrivSqZeroExt.inrHom R N).comp f)
+  · intro x y
+    simp only [LinearMap.coe_comp, Function.comp_apply, TrivSqZeroExt.inrHom_apply,
+      TrivSqZeroExt.inr_mul_inr]
+  · intro r x
+    simp only [map_smul, LinearMap.coe_comp, Function.comp_apply, TrivSqZeroExt.inrHom_apply,
+      TrivSqZeroExt.inlAlgHom_apply]
+    rw [TrivSqZeroExt.inl_mul_eq_smul]
+  · intro r x
+    simp only [op_smul_eq_smul, map_smul, LinearMap.coe_comp, Function.comp_apply,
+      TrivSqZeroExt.inrHom_apply, TrivSqZeroExt.inlAlgHom_apply]
+    rw [TrivSqZeroExt.mul_inl_eq_op_smul]
+    simp only [op_smul_eq_smul]
+
+theorem snd_comp_mapTriv [Module Rᵐᵒᵖ M] [IsCentralScalar R M] [Module Rᵐᵒᵖ N] [IsCentralScalar R N]
+    (f : M →ₗ[R] N) : (TrivSqZeroExt.sndHom R N).comp (mapTriv f).toLinearMap
+    = f.comp (TrivSqZeroExt.sndHom R M) := by
+  unfold mapTriv
+  apply TrivSqZeroExt.linearMap_ext
+  · intro r
+    simp only [LinearMap.coe_comp, Function.comp_apply, AlgHom.toLinearMap_apply,
+      TrivSqZeroExt.lift_apply_inl, TrivSqZeroExt.inlAlgHom_apply, TrivSqZeroExt.sndHom_apply,
+      TrivSqZeroExt.snd_inl, map_zero]
+  · intro x
+    simp only [LinearMap.coe_comp, Function.comp_apply, AlgHom.toLinearMap_apply,
+      TrivSqZeroExt.lift_apply_inr, TrivSqZeroExt.inrHom_apply, TrivSqZeroExt.sndHom_apply,
+      TrivSqZeroExt.snd_inr]
```

Comment:
These two results should move to `TrivSqZeroExt.lean`

Reply:
I moved the def and theorem. Morally these should also be done for a noncommutative base ring, but it would require `f` to be `R`-bilinear, not just `R`-linear, so it might complicate things.